### PR TITLE
Cleanup non-namespaced objects in e2e test during interrupts

### LINKED ISSUE #96022: Cleanup non-namespaced objects in interrupted e2e tests
/sig storage

**What happened**:
When you interrupt the e2e storage stress test, namespaced objects are cleaned up, but non-namespaced objects like StorageClasses are not cleaned up.

**What you expected to happen**:
Non-namespaced objects like StorageClasses are cleaned up so long as the user doesn't interrupt a second time.

**How to reproduce it (as minimally and precisely as possible)**:
```
# Make the e2e.test binary.
make WHAT=test/e2e/e2e.test

# Run the stress test, I chose a specific storage driver so there's only one spec.
# Assumes you already have a testdriver.yaml setup.
_output/bin/e2e.test \
	--kubeconfig=$HOME/.kube/config \
	-ginkgo.focus='External Storage \[Driver\: ebs\.csi\.aws\.com\] \[Testpattern\: Dynamic PV \(default fs\)\] stress' \
	-storage.testdriver=testdriver.yaml

# Ctrl-C the test while it's running

# Check what StorageClasses exist after the binary exits.
kubectl get sc
```

### PR DESCRIPTION
**Pull request by chrishenzie**
/sig storage

**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
This PR fixes an issue where if this e2e test is interrupted, only the namespace gets cleaned up. Non-namespaced objects like StorageClasses are not cleaned up.

**Which issue(s) this PR fixes**:

Fixes #96022

**Special notes for your reviewer**:

The test framework has an `AddAfterEach()` block that allows us to handle this test cleanup logic.

Since we are handling the test cleanup in an `AfterEach()` step, I figured it was intuitive to handle the equivalent test setup in a `BeforeEach()` step. I split this into two parts. One that asserts things before the test framework is created (which creates a namespace), and one that sets up everything else after. These `BeforeEach()` blocks run sequentially before every `It()` block.

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

@msau42

### CODE REVIEW

**Comment by msau42** [comment]
/ok-to-test
/assign @gnufied 
I thought we already did this?

File: test/e2e/storage/testsuites/stress.go [hunk 1]
```diff
@@ -141,27 +157,16 @@ func (t *stressTestSuite) DefineTests(driver TestDriver, pattern testpatterns.Te
...
-			pod, err := e2epod.MakeSecPod(&podConfig)
-			framework.ExpectNoError(err)
+	ginkgo.BeforeEach(func() {
+		init()
+		createPodsAndVolumes()
+	})
 
-			l.pods = append(l.pods, pod)
-		}
+	f.AddAfterEach("cleanup", func(f *framework.Framework, failed bool) {
```
Comment (msau42) [inline test/e2e/storage/testsuites/stress.go:165]: Do you know what the difference is between `AddAfterEach` and `ginkgo.AfterEach`?

Comment (chrishenzie) [reply test/e2e/storage/testsuites/stress.go:165]: `AddAfterEach` appends a function to execute as part of the framework's `AfterEach`, which is registered with ginkgo via `ginkgo.AfterEach`:
https://github.com/kubernetes/kubernetes/blob/7bf9b0a3beca74dabb85f1d0034ad78f689f807c/test/e2e/framework/framework.go#L175

I expected I could replace this with a call to `ginkgo.AfterEach` and see it work, but when I tested it, it didn't run.

**Pull request merged**